src/realm/object-store/impl/collection_change_builder.cpp

```diff
@@ -25,10 +25,46 @@
 using namespace realm;
 using namespace realm::_impl;
 
+CollectionChangeBuilder::CollectionChangeBuilder(CollectionChangeBuilder&& other)
```

Comment:
These shouldn't need to be overridden. I'm assuming it's because there's something which depends on collection_root_was_deleted being false in a moved-from CCB, but that's generally a bad thing to do and instead anything which cares about the state of a moved-from object should explicitly set it.